tests/event.spec.ts

```diff
@@ -1,101 +1,111 @@
-
-        // Fetch the events after the comment is created
-        const events = await api(request).get("/api/events");
-
-        // Filter the event related to the newly created comment
-        const event = events.reverse().find(item => (item.entityId === newCommentId && item.entityType === "Comment"));
-
-        // Schema Checks
-        expect(event).toBeDefined();  // Ensure the event exists
-        expect(event.action).toBe('Commented');  // Check the action is "Commented"
-        expect(event.entityName).toBe('This is a test comment');  // The entityName should be the comment text
-        expect(event.content).toBe("This is a test comment");  // Check the event content
-        expect(event.entityType).toBe('Comment');  // Ensure the entityType is "Comment"
-        expect(event.entityId).toBe(newCommentId);  // The entityId should be the ideaId related to the comment
-        expect(event.eventUser.username).toBe(config.user.username);  // Check the eventUser is the comment author
+    // Fetch all events
+    const events = await api(request).get("/api/events");
+
+    // Find the event related to the updated idea status
+    const event = events
+      .reverse()
+      .find(
+        (item) =>
+          item.entityId === updatedIdea.id && item.entityType === "IdeaCard"
+      );
+
+    // Schema Checks
+    expect(event).toBeDefined(); // Ensure the event exists
+    expect(event.action).toBe("Idea Updated"); // Check the action
+    expect(event.entityName).toBe("TestEventIdea"); // Check the entity name
+    expect(event.content).toBe("Keep testing"); // Check the content format
+  });
+
+  test("Create new event on comment create", async ({ request }) => {
+    // Create a new comment on the existing idea
+    await api(request).post("/api/comments", {
+      idea_card: { id: ideaId }, // Associate the comment with the created idea
+      text: "This is a test comment", // Comment content
     });
 
-});
\ No newline at end of file
+    // Fetch the events after the comment is created
+    const events = await api(request).get("/api/events");
+
+    // Filter the event related to the newly created comment
+    const event = events
+      .reverse()
+      .find(
+        (item) => item.entityId === ideaId && item.entityType === "Comment"
+      );
+
+    // Schema Checks
+    expect(event).toBeDefined(); // Ensure the event exists
+    expect(event.action).toBe("Commented"); // Check the action is "Commented"
+    expect(event.entityName).toBe("This is a test comment"); // The entityName should be the comment text
+    expect(event.content).toBe("This is a test comment"); // Check the event content
+    expect(event.entityType).toBe("Comment"); // Ensure the entityType is "Comment"
+    expect(event.entityId).toBe(ideaId); // The entityId should be the ideaId related to the comment
+    expect(event.eventUser.username).toBe(config.user.username); // Check the eventUser is the comment author
```

Comment:
Will there also be an event for the idea owner, to notify someone commented on their idea?

Reply:
@chungthuang no new event is created for the idea owner, when this **comment event** is created, the user and all who are already _subscribed_ to the idea (idea owner, users who commented) get notified

Reply:
Ok that makes sense. Can this test verify that both the idea owner and commenter get notified?

Reply:
@chungthuang no because the comment is made by the idea owner and he cant get notified multiple time for the same event

Reply:
That sounds like we should use a different user to comment to make this test more realistic?

Reply:
yes we can login with **user2** and comment and then fetch the notifications resulting from that event